### ISSUE TITLE
Stubs are ignored

### ISSUE THREAD

**Issue by zmitic** [klesun/deep-assoc-completion#162]
First; thanks for this plugin. Autocomplete works absolutely amazing even with generics (that I use a **lot**).

The main problem I noticed is that stubs I have (for Symfony forms) do not work. But with LSP plugin, they work. I even disabled it to make sure there are no conflicts.

This is real stub (shortened):

```php
/**
 * @template T
 */
interface FormInterface extends \ArrayAccess, \Traversable, \Countable
{
    /**
     * @psalm-return T|null
     */
    public function getData();
```

and controller (I left only the important parts):

```php
    /**
     * @param FormInterface<Contract> $form
     */
    public function create(FormInterface $form): Response
    {
           $contract = $form->getData();
           $contract->  // nothing here
    }
```

Do I need to configure it somewhere?

**Comment by klesun** [comment]
Hi. Thanks for the report, seems like I broke `@template` support in some of the recent releases. Will take a look tonight.

P.S. may I ask, what LSP are you using? I did not know there was something decent for static typing in phpstorm...

**Comment by zmitic** [comment]
@klesun I use this in PHPStorm: https://psalm.dev/docs/running_psalm/language_server/

But it does have few problems; selecting suggestion from generic param is duplicated (undo removes the other one), can't ctrl+click to jump to method, not so easy to start it on bigger projects (user has to close all files and restart PHPStorm)...

Using array objects does not offer suggestions but at least it shows error if I miss type it.

Probably few more but I guess I am too used.

---

But today I tested your plugin and generic value returned suggestions instantly for this use-case (shortened):

```php
/**
 * @template T
 */
class LazyValue
{
    /**
     * @psalm-var Closure(): T
     */
    private Closure $callable;

    /**
     * @psalm-var T
     * @psalm-suppress PropertyNotSetInConstructor
     */
    private $value;

    /**
     * @param Closure(): T $callable
     */
    public function __construct(Closure $callable)
    {
        $this->callable = $callable;
    }
    /**
     * @psalm-return T
     */
    public function getValue() // runs closure once, caches the value and returns it
}
```

The caller:

![image](https://user-images.githubusercontent.com/1964158/89303035-e92b5f80-d66b-11ea-8a56-9a9bbdced027.png)


---

So it is not that templates are not working, only the stubs. Can it be because of duplication? My stubs are in ``/stubs`` folder, original files are in ``/vendor`` but they all have same signature including namespace.

**Comment by klesun** [comment]
Hm, duplicates could indeed possibly be the cause. Checking...

**Comment by klesun** [comment]
I found several problems in the plugin:
- `@psalm-return` does not work, probably I don't parse it correctly because it's contents are not highlighted
- class `@template` generics don't seem to be interpreted for right class (current instead of the one in which method is located)

Guess it's a good opportunity to revive the development for a few weeks...

**Comment by zmitic** [comment]
LazyValue class uses ``@psalm-return`` and it works flawless.

Later in the day I will try to exclude original files and see if having stubs only would help.

**Comment by klesun** [comment]
Not sure, but I think that LazyValue works because it infers type info from `private $value;` (given that you return it somewhere inside the `getValue` implementation)

**Comment by klesun** [comment]
As a temporary fix, setting _File -> Settings -> Languages & Frameworks -> PHP -> deep-assoc-completion -> Pass args to interface implementations_ to true should fix the issue I think.

I'll think whether performance decrease in framework-heavy projects may be avoided with making generics work in this use case without setting the flag...

**Comment by klesun** [comment]
Fixed in [2020.08.18.001](https://plugins.jetbrains.com/plugin/9927-deep-assoc-completion/versions/stable/94762)

![image](https://user-images.githubusercontent.com/5202330/90449042-448b3200-e0ef-11ea-9b78-7cfb566cefa1.png)

(please, let me know, if this happen to not solve your issue)

**Comment by zmitic** [comment]
It works!

![image](https://user-images.githubusercontent.com/1964158/90450473-1b1bd780-e0ea-11ea-9482-025cd37c8570.png)


Although it doesn't work when ``psalm-`` prefix is used

![image](https://user-images.githubusercontent.com/1964158/90450526-3edf1d80-e0ea-11ea-943b-bdf105b4ca3b.png)



it doesn't matter; I will be soon changing all my stubs and see how it works.

---

Thank you **a lot**

**Comment by klesun** [comment]
Oh, right, thanks, almost forgot there was one more issue with `@psalm-return` parsing

**Comment by zmitic** [comment]
There is one minor issue, probably irrelevant. Because of nested-set requiring its own parent repository, I put code in traits like this:

```php
class CityRepository extends ServiceEntityRepository
{
    /** @use RepositoryUtilsTrait<City> */
    use RepositoryUtilsTrait;

    public function __construct(ManagerRegistry $registry)
    {
        parent::__construct($registry, City::class);
    }
}
```

Trait itself:

```php
    /**
     * @return ?T
     */
    public function getOneResultOrNull(Generator ...$generators)
    {
       // ...
    }


```

No results:

![image](https://user-images.githubusercontent.com/1964158/90451401-06404380-e0ec-11ea-8a7a-fec5cfab19aa.png)

**Comment by klesun** [comment]
Wow, did not even know PSALM has format for delegating generics to traits.

Filed in a separate issue: https://github.com/klesun/deep-assoc-completion/issues/164

**Comment by klesun** [comment]
Fixed `@psalm-return` in [2020.08.18.002](https://plugins.jetbrains.com/plugin/9927-deep-assoc-completion/versions/stable/94773)

![image](https://user-images.githubusercontent.com/5202330/90455415-adc67180-e0fe-11ea-9352-4ebcddca2cff.png)
